### PR TITLE
Add `iterableEquals`, `listEquals`, etc. extension methods.

### PR DESCRIPTION
**Pull request by lrhn**
Also fixes type issue with `UnorderedIterableEquality`.
FYI @johnniwinther

### CODE REVIEW

File: lib/src/equality.dart [hunk 1]
```diff
@@ -272,7 +272,7 @@ class UnorderedIterableEquality<E> extends _UnorderedEquality<E, Iterable<E>?> {
 /// The [equals] and [hash] methods accepts `null` values,
 /// even if the [isValidKey] returns `false` for `null`.
 /// The [hash] of `null` is `null.hashCode`.
-class SetEquality<E> extends _UnorderedEquality<E, Set<E>?> {
+class SetEquality<E> extends _UnorderedEquality<E, Set<E>> {
```
Comment (lrhn) [inline lib/src/equality.dart:275]: This made `SetEquality<int>` implement `Equality<Set<int>?>`, which means that writing:
```dart
Equality<Set<X>> eq = SetEquality<X>(xEquality);
```
would be a type error. Seems nobody hit that yet, but fixing it to be certain.

(We still let `SetEquality<X>` accept `Set<X>?` as arguments to equality, but that's just being permissive in what it accepts, not in what it *is*).

Comment (natebosch) [reply lib/src/equality.dart:275]: Could we handle this change in a separate PR?

File: lib/src/iterable_extensions.dart [hunk 2]
```diff
@@ -18,6 +19,38 @@ import 'algorithms.dart';
...
 /// [IterableComparableExtension] and [IterableNullableExtension].
 extension IterableExtension<T> on Iterable<T> {
+  /// Whether this iterable has the same element order as [elements].
+  ///
+  /// Checks whether this iterable has the same elements as [elements],
+  /// in the same iteration order.
+  ///
+  /// Uses [equality] for determining if the values are the same.
+  /// If [equality] is omitted, the default `==` behavior is used.
+  bool iterableEquals(Iterable<T> elements, [Equality<T>? equality]) {
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:29]: This reads strangely to me. What do you think of `equalsIterable` instead?

File: lib/src/iterable_extensions.dart [hunk 3]
```diff
@@ -18,6 +19,38 @@ import 'algorithms.dart';
 /// iterables with specific element types include those of
 /// [IterableComparableExtension] and [IterableNullableExtension].
 extension IterableExtension<T> on Iterable<T> {
+  /// Whether this iterable has the same element order as [elements].
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:22]: I think this should read "same values and same order as"

File: lib/src/iterable_extensions.dart [hunk 4]
```diff
@@ -18,6 +19,38 @@ import 'algorithms.dart';
...
+  ///
+  /// Checks whether this iterable has the same elements as [elements],
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:24]: This feels like a duplicate sentence.

File: lib/src/iterable_extensions.dart [hunk 5]
```diff
@@ -18,6 +19,38 @@ import 'algorithms.dart';
...
+  }
+
+  /// Whether this iterable has the same unordered elements as [elements].
+  ///
+  /// Checks whether this iterable has the same elements as [elements],
+  /// but not necessarily in the same iteration order.
+  ///
+  /// Uses [equality] for determining if the values are the same.
+  /// If [equality] is omitted, the default `==` behavior is used.
+  bool iterableUnorderedEquals(Iterable<T> elements, [Equality<T>? equality]) {
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:48]: How about `unorderedEqualsIterable`

File: lib/src/iterable_extensions.dart [hunk 6]
```diff
@@ -793,3 +826,66 @@ extension ComparatorExtension<T> on Comparator<T> {
         return result;
       };
 }
+
+/// Extensions on sets.
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:830]: [nit] This comment can be omitted unless we have something more interesting to add.

File: lib/src/iterable_extensions.dart [hunk 7]
```diff
@@ -793,3 +826,66 @@ extension ComparatorExtension<T> on Comparator<T> {
...
+  /// Whether this set has the same unordered elements as [elements].
+  ///
+  /// Checks whether this set has the same elements as [elements],
+  /// but not necessarily in the same iteration order.
+  ///
+  /// Uses [equality] for determining if the values are the same.
+  /// If [equality] is omitted, default to the equality used by
+  /// this set.
+  /// _(Notice: This differs from [SetEquality],
+  /// which is symmetric in the arguments)_.
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:841]: I don't think I understand this statement, and it doesn't look like the `SetEquality` docs mention "symmetric". Can you expand this?

**Comment by mit-mit** [comment]
Closing as assumed stale

**Comment by lrhn** [comment]
Doesn't seem fixed. The `SetEquality` type problem is still there.

**Pull request closed without merging**